File: src/services/availability.service.js
import api from './api'

const availabilityService = {
  // Public - Check if a time slot is available
  checkAvailability: async (date, startTime, endTime, adminId) => {
    const response = await api.get(`/availability`, {
      params: { date, startTime, endTime, adminId }
    })
    return response.data.data || response.data
  },

  lockSlot: async (adminId, eventDate, startTime, endTime) => {
    const response = await api.post(`/slots/lock`, { adminId, eventDate, startTime, endTime })
    return response.data
  },

  unlockSlot: async (adminId, eventDate) => {
    const response = await api.post(`/slots/unlock`, { adminId, eventDate })
    return response.data
  }
}

export default availabilityService

File: OneDrive/Desktop/EventBookingApp/src/main/java/com/ranadj/service/AvailabilityService.java
package com.ranadj.service;

import com.ranadj.dto.AvailabilityResponse;

import java.time.LocalDate;
import java.time.LocalTime;

/**
 * Service interface for checking DJ availability.
 */
public interface AvailabilityService {

    /**
     * Checks if a booking slot is available (i.e. has no overlapping CONFIRMED bookings).
     */
    AvailabilityResponse checkAvailability(LocalDate date, LocalTime startTime, LocalTime endTime);
}
